Power.lua

```diff
@@ -1,21 +1,21 @@
-local Power, parent = torch.class('nn.Power','nn.Module')
-
-function Power:__init(p)
-   parent.__init(self)
-   self.pow = p
-   if not p then
-      error('nn.Power(power)')
-   end
-end
-
-function Power:updateOutput(input)
-   self.output:resizeAs(input):copy(input)
-   self.output:pow(self.pow)
-   return self.output
-end
-
-function Power:updateGradInput(input, gradOutput)
-   self.gradInput:resizeAs(input):copy(gradOutput)
-   self.gradInput:cmul(self.output):cdiv(input):mul(self.pow)
-   return self.gradInput
-end
+local Power, parent = torch.class('nn.Power','nn.Module')
+
+function Power:__init(p)
+   parent.__init(self)
+   self.pow = p
+   if not p then
+      error('nn.Power(power)')
+   end
+end
+
+function Power:updateOutput(input)
+   self.output:resizeAs(input):copy(input)
+   self.output:pow(self.pow)
+   return self.output
+end
+
+function Power:updateGradInput(input, gradOutput)
+   self.gradInput:resizeAs(input):copy(gradOutput)
+   self.gradInput:cmul(torch.pow(input,self.pow-1)):mul(self.pow)

```

Comment:
torch.pow(input, self.pow-1) will create a newly allocated tensor every single time as a temporary buffer. This is bad practice and will also give  a bad slow-down on GPU.

So instead, have a temporary buffer:
```lua
self.buffer = self.buffer or input.new()
self.buffer:resizeAs(input):copy(input)
self.buffer:pow(self.pow-1)
self.gradInput:cmul(self.buffer):mul(self.pow)
```